### PR TITLE
Replace SLAM toolbox by Cartographer in TIAGo example

### LINKED ISSUE #635: Drop slam-toolbox for other more robust packages
The slam-toolbox package is used in the [Tiago tutorial](https://github.com/cyberbotics/webots_ros2/wiki/Example-TIAGo) for SLAM. There have been reports that the package can be inaccurate when used with the diff-drive controller (see https://github.com/cyberbotics/webots/issues/5540#issuecomment-1371748895). Here is the kind of map generated with the Tiago:

Screenshot from 2023-02-20 17-23-04

As such, I suggest to replace the package with a more accurate alternative like AMCL or Cartographer in the Tiago tutorial. Cartographer is already used in the [Turtlebot tutorial](https://github.com/cyberbotics/webots_ros2/wiki/SLAM-with-TurtleBot3). @delihus has also demonstrated that AMCL gives better results, in the scope of a project using the ROSbot.

The plan is to transfer tutorials to the official ROS 2 documentation. If there is no opposition, slam-toolbox will be replaced by another package in the new Tiago tutorial.

### PR DESCRIPTION
**Pull request by ygoumaz**
Resolves #635.

The cartographer node is responsible for the SLAM computation.
The cartographer grid node allows to display visually the map in RViz.

### CODE REVIEW

File: webots_ros2/CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,7 @@ Changelog for package webots_ros2
 2023.0.3 (2023-XX-XX)
 ------------------
 * Improved the navigation of the Tiago example.
+* Replaced Slam-Toolbox by Cartographer in the Tiago example.
```
Comment (stefaniapedrazzi) [inline webots_ros2/CHANGELOG.rst:8]: Just for info, the correct writing of the robot name is TIAGo.

**Review by ad-daniel — CHANGES_REQUESTED** [review]
I'm getting this error on foxy:
```
ages/launch/launch_service.py:226> exception=SubstitutionFailure("executable 'cartographer_occupancy_grid_node' not found on the libexec directory '/opt/ros/foxy/lib/cartographer_ros' ")>
Traceback (most recent call last):
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 228, in _process_one_event
    await self.__process_event(next_event)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_service.py", line 248, in __process_event
    visit_all_entities_and_collect_futures(entity, self.__context))
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
    sub_entities = entity.visit(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/action.py", line 108, in visit
    return self.execute(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch_ros/actions/node.py", line 453, in execute
    ret = super().execute(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/execute_process.py", line 823, in execute
    self.__expand_substitutions(context)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/execute_process.py", line 668, in __expand_substitutions
    cmd = [perform_substitutions(context, x) for x in self.__cmd]
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/actions/execute_process.py", line 668, in <listcomp>
    cmd = [perform_substitutions(context, x) for x in self.__cmd]
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/perform_substitutions_impl.py", line 26, in perform_substitutions
    return ''.join([context.perform_substitution(sub) for sub in subs])
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/utilities/perform_substitutions_impl.py", line 26, in <listcomp>
    return ''.join([context.perform_substitution(sub) for sub in subs])
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch/launch_context.py", line 232, in perform_substitution
    return substitution.perform(self)
  File "/opt/ros/foxy/lib/python3.8/site-packages/launch_ros/substitutions/executable_in_package.py", line 84, in perform
    raise SubstitutionFailure(
launch.substitutions.substitution_failure.SubstitutionFailure: executable 'cartographer_occupancy_grid_node' not found on the libexec directory '/opt/ros/foxy/lib/cartographer_ros' 
[INFO] [rviz2-8]: process started with pid [1741686]
[INFO] [cartographer_node-9]: process started with pid [1741688]

```

**Comment by ad-daniel** [comment]
a

**Review by ad-daniel — APPROVED** [review]
Worked both on humble and foxy (at least what is supported), thank you

**Pull request merged**